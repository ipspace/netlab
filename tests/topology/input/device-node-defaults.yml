#
# Test provider-specific node defaults
#

provider: clab

nodes: 
  h1:
    device: linux
  h2:
    device: linux
  rtr:
    id: 1
    device: cumulus
    module: [ ospf ]
    libvirt.image: CumulusCommunity/cumulus-vx:5.2.0

links:
- h1:
  rtr:
- h2:
  rtr:
