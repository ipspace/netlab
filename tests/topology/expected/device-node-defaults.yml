input:
- topology/input/device-node-defaults.yml
- package:topology-defaults.yml
links:
- bridge: input_1
  gateway:
    ipv4: 172.16.0.1/24
  interfaces:
  - ipv4: 172.16.0.2/24
    node: h1
  - ipv4: 172.16.0.1/24
    node: rtr
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 172.16.0.0/24
  role: stub
  type: lan
- bridge: input_2
  gateway:
    ipv4: 172.16.1.1/24
  interfaces:
  - ipv4: 172.16.1.3/24
    node: h2
  - ipv4: 172.16.1.1/24
    node: rtr
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 172.16.1.0/24
  role: stub
  type: lan
module:
- ospf
name: input
nodes:
  h1:
    af:
      ipv4: true
    box: python:3.9-alpine
    clab:
      kind: linux
    device: linux
    hostname: clab-input-h1
    id: 2
    interfaces:
    - bridge: input_1
      gateway:
        ipv4: 172.16.0.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.2/24
      linkindex: 1
      mtu: 1500
      name: h1 -> [rtr]
      neighbors:
      - ifname: swp1
        ipv4: 172.16.0.1/24
        node: rtr
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    mtu: 1500
    name: h1
    role: host
  h2:
    af:
      ipv4: true
    box: python:3.9-alpine
    clab:
      kind: linux
    device: linux
    hostname: clab-input-h2
    id: 3
    interfaces:
    - bridge: input_2
      gateway:
        ipv4: 172.16.1.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.3/24
      linkindex: 2
      mtu: 1500
      name: h2 -> [rtr]
      neighbors:
      - ifname: swp2
        ipv4: 172.16.1.1/24
        node: rtr
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    mtu: 1500
    name: h2
    role: host
  rtr:
    af:
      ipv4: true
    box: networkop/cx:4.3.0
    clab:
      kind: cvx
    device: cumulus
    hostname: clab-input-rtr
    id: 1
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: swp1
      ipv4: 172.16.0.1/24
      linkindex: 1
      mtu: 1500
      name: rtr -> [h1]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.0.2/24
        node: h1
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: true
      role: stub
      type: lan
    - bridge: input_2
      ifindex: 2
      ifname: swp2
      ipv4: 172.16.1.1/24
      linkindex: 2
      mtu: 1500
      name: rtr -> [h2]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.3/24
        node: h2
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: true
      role: stub
      type: lan
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - ospf
    mtu: 1500
    name: rtr
    ospf:
      area: 0.0.0.0
      router_id: 10.0.0.1
    runtime: docker
ospf:
  area: 0.0.0.0
provider: clab
